### PR TITLE
Request response operations

### PR DESCRIPTION
**Pull request by bretambrose**
Basic support for submitting request operations and creating streaming operations.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/request-response/request_response_client.c [hunk 1]
```diff
@@ -397,3 +806,390 @@ struct aws_mqtt_request_response_client *aws_mqtt_request_response_client_releas
...
+
+    storage->options.response_paths = storage->operation_response_paths.data;
+}
+
+int aws_mqtt_request_response_client_submit_request(
+    struct aws_mqtt_request_response_client *client,
+    const struct aws_mqtt_request_operation_options *request_options) {
+
+    if (client == NULL) {
+        return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
```
Comment (sbSteveK) [inline source/request-response/request_response_client.c:1113]: Trivial: Should we log the reason for this error being raised here?

File: source/request-response/request_response_client.c [hunk 2]
```diff
@@ -397,3 +806,390 @@ struct aws_mqtt_request_response_client *aws_mqtt_request_response_client_releas
...
+    AWS_FATAL_ASSERT(
+        aws_byte_buf_append_and_update(&storage->operation_data, &storage->options.topic_filter) == AWS_OP_SUCCESS);
+}
+
+struct aws_mqtt_rr_client_operation *aws_mqtt_request_response_client_create_streaming_operation(
+    struct aws_mqtt_request_response_client *client,
+    const struct aws_mqtt_streaming_operation_options *streaming_options) {
+
+    if (client == NULL) {
+        aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
```
Comment (sbSteveK) [inline source/request-response/request_response_client.c:1154]: Trivial: same as above. Should we log the client null is the invalid argument?

**Review by sbSteveK — APPROVED** [review]
Trivial comments may need attending at discretion.

File: include/aws/mqtt/request-response/request_response_client.h [hunk 3]
```diff
@@ -6,11 +6,69 @@
...
 
 struct aws_mqtt_request_response_client;
 struct aws_mqtt_client_connection;
 struct aws_mqtt5_client;
+struct aws_mqtt_streaming_operation;
+
+struct aws_mqtt_request_operation_response_path {
+    struct aws_byte_cursor topic;
+
+    /* potential point of expansion into an abstract "extractor" if we ever need to support non-JSON payloads */
```
[the remark this answers is not in the document]
Comment (bretambrose) [reply include/aws/mqtt/request-response/request_response_client.h:21]: No longer part of validity; 0 length is fine and means no correlation token involved

File: source/request-response/request_response_client.c [hunk 4]
```diff
@@ -13,6 +13,307 @@
...
+
+enum aws_mqtt_request_response_operation_type {
+    AWS_MRROT_REQUEST,
+    AWS_MRROT_STREAMING,
+};
+
+enum aws_mqtt_request_response_operation_state {
+    AWS_MRROS_NONE,                 // creation -> in event loop enqueue
+    AWS_MRROS_QUEUED,               // in event loop queue -> non blocked response from subscription manager
+    AWS_MRROS_PENDING_SUBSCRIPTION, // subscribing response from sub manager -> subscription success/failure event
```
[the remark this answers is not in the document]
Comment (bretambrose) [reply source/request-response/request_response_client.c:26]: Moved pending destroy but reluctant to rename with streaming/request

File: source/request-response/request_response_client.c (hunk 1 above)
Comment (bretambrose) [reply source/request-response/request_response_client.c:1113]: It's something that should never happen and would indicate that our bindings are broken which I don't think will be helpful in the long run.

File: source/request-response/request_response_client.c (hunk 2 above)
Comment (bretambrose) [reply source/request-response/request_response_client.c:1154]: Same response

**Pull request closed without merging**